examples/kops-with-vpc/README.md

```diff
@@ -0,0 +1,184 @@
+# Example to test the Kops workflow with an existing VPC on AWS
+
+## Summary
+
+We first use `terraform` to create:
+1. a VPC,
+2. it's subnets and
+3. an internet gateway, along with route tables for local and internet traffic.
+
+We then use `kops` to create the kubernetes cluster in this existing VPC, where `kops`, by itself, manages the creation of:
+1. DNS records under the base FQDN,
+2. The auto-scaling group, and its launch configurations
+3. EBS volumes
+4. IAM instance profiles, role policies
+5. Keypairs for every kubernetes component
+6. Files and secrets required to configu kubernetes clusters
+7. Security groups
+...and modifies the tags on existing subnets to make things easier for discovery within the kubernetes cluster
+
+## Environment creation and deployment
+`NOTE`: For usage with `aws-env` have each command below execute in the form `aws-env -p <your-profile> -- <the-command>`.
+
+### Part 1: Terraform
+
+```
+pushd terraform
+source prod/env
+```
+
+1. Create the VPC
+    - 1 SSH Keypair to use throughout
+    - 1 VPC, and
+    - 3 subnets.
+
+```
+make generate-ssh-keypair
+make import-ssh-keypair
+make init
+make init-plan
+make apply
+```
+
+2. Create the NAT Gateway
+    - the Internet Gateway, and
+    - its route tables, for traffic between the 3 subnets and the public internet.
+
+```
+make plan
+make apply
+```
+
+### Part 2: Kubernetes (Kops)
+
+```
+popd; pushd kubernetes
+source prod/env
+cp ../terraform/*.pem* prod/    # copy the SSH Keypair
+make create-s3-bucket           # create the S3 bucket for `kops` state
+```
+
+1. Create the cluster configuration
+```
+make kops-create-cluster
+```
+2. Edit the cluster configuration to help `kops` use the existing VPC and its subnets.
+```
+make kops-edit-cluster
+```
+...here is a table of variable names between the `terraform` output and `kops` configuration that should match after editing the `kops` configuration.
+
+|Terraform output|Kops configuration|
+|----------------|------------------|
+|`vpc_id`|`networkID`|
```

Comment:
Since the VPC ID and VPC CIDR are passed to `kops create cluster` from the `env` file, these should not need to be changed here.